File: frontend/src/api/video_task_config.js
import http from './http'

const BASE = '/video-task-config'

/**
 * @returns {Promise<{ top_percent: number, discard_below: number, select_percent: number,
 *   auto_publish_enabled: boolean, auto_publish_model: string, auto_publish_prompt: string }>}
 */
export async function fetchTaskConfig() {
  const { data } = await http.get(BASE)
  return data
}

/**
 * @param {Object} payload
 * @param {number} [payload.top_percent]
 * @param {number} [payload.discard_below]
 * @param {number} [payload.select_percent]
 * @param {boolean} [payload.auto_publish_enabled]
 * @param {string} [payload.auto_publish_model]
 * @param {string} [payload.auto_publish_prompt]
 */
export async function updateTaskConfig(payload) {
  const { data } = await http.put(BASE, payload)
  return data
}

File: backend/app/schemas/video_task_config.py
"""Schemas for video task pipeline and publish pool configuration."""
from __future__ import annotations

from pydantic import BaseModel, Field


class VideoTaskConfigRead(BaseModel):
    """Read view of video task config (per-owner singleton)."""

    top_percent: float = 30.0       # Step 1: 直接挑选前 N%
    discard_below: float = 40.0     # Step 2: 丢弃分数低于此值的
    select_percent: float = 50.0    # Step 3: 剩下的再选前 Y%

    auto_publish_enabled: bool = False
    auto_publish_model: str = "gemini-3.1-pro-preview"
    auto_publish_prompt: str = ""


class VideoTaskConfigUpdate(BaseModel):
    """Update payload for video task config."""

    top_percent: float | None = Field(default=None, ge=0, le=100)
    discard_below: float | None = Field(default=None, ge=0, le=100)
    select_percent: float | None = Field(default=None, ge=0, le=100)

    auto_publish_enabled: bool | None = None
    auto_publish_model: str | None = None
    auto_publish_prompt: str | None = None
